lib/log_parser.rb

```diff
@@ -1,16 +1,30 @@
+require 'json'
+
 class Log_Parser
-  attr_accessor :path
+  attr_accessor :path, :file_data
 
   def initialize(path)
     @path = path
+    @file_data = self.openfile
```

Comment:
```suggestion
    @file_data = openfile
```

Não precisa declarar o self aqui pois o próprio ruby implicitamente coloca